pkg/gce-pd-csi-driver/node.go

```diff
@@ -51,6 +51,10 @@ type GCENodeServer struct {
 	// for that same volume (as defined by VolumeID) return an Aborted error
 	volumeLocks *common.VolumeLocks
 
+	// deviceInUseErrors keeps tracks of device names and a timestamp for when an error is
+	// encounted for that device
+	deviceInUseErrors map[string]time.Time
```

Comment:
There may be concurrent CSI API calls. Can you migrate this to an atomic data structure, or add a mechanism to serialize data access?

See VolumeLocks for inspiration: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/99542b07e40b599c4f9106cd96ff1ebeb577351a/pkg/common/volume_lock.go#L31

Reply:
Done. Also moved deviceInUseErrors struct to new file: pkg/gce-pd-csi-driver/device_error_map.go